src/Microsoft.TestPlatform.Client/Execution/TestRunRequest.cs

```diff
@@ -659,6 +660,17 @@ public int LaunchProcessWithDebuggerAttached(TestProcessStartInfo testProcessSta
             return processId;
         }
 
+        /// <inheritdoc />
+        public bool AttachDebuggerToProcess(int pid)
+        {
+            if (!(this.testRunCriteria.TestHostLauncher is ITestHostLauncher2))
+            {
+                return false;
+            }
+
+            return (this.testRunCriteria.TestHostLauncher as ITestHostLauncher2).AttachDebuggerToProcess(pid);
```

Comment:
This will cast twice, once for the `is` check and once for the `as` check. The `is` check allows you to save the result of the check into a variable that is typed as the type operand in the `is` check. I would also negate the check to avoid the mental overhead of negating the condition in your head:

```cs
if (this.testRunCriteria.TestHostLauncher is ITestHostLauncher2 launcher)
{
    return launcher.AttachDebuggerToProcess(pid);
}

return false;
```

Reply:
Thanks for the tip. 👍 
I refactored the code to use 'as' instead of 'is' to avoid double casting. I further refactored it to look like this: 
```
var launcher = this.testRunCriteria.TestHostLauncher as ITestHostLauncher2;

return (launcher != null) ? launcher.AttachDebuggerToProcess(pid) : false;
```

However, if I wouldn't have used the conditional operator I would've still preferred to have the negation. The reason for that is that I am trying to follow an early return pattern (https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement) where you get the incompatible conditions quickly out of the way. After that you're only left with code that's actually useful. I know it's not the case for now, however in the future we might have to add more sanity checks to these calls and I prefer to have them all on top as I feel they're easier to read this way.

Reply:
imho this refactoring is worse than the negation + is.

Reply:
Done.